toolkit/components/uniffi-bindgen-gecko-js/fixtures/tests/test_rondpoint.js

```diff
@@ -3,25 +3,232 @@
+   //      we show that lowering from js and lifting into rust has values that both js and rust
+   //      both stringify in the same way. i.e. the same values.
+   //
+   // If we roundtripping proves the symmetry of our lowering/lifting from here to rust, and lowering/lifting from rust to here,
+   // and this convinces us that lowering/lifting from here to rust is correct, then 
+   // together, we've shown the correctness of the return leg.
+   const st = await Stringifier.init();
+
+   const affirmEnchaine = async (arr, fn) => {
+      for (const member of arr) {
+         Assert.equal(await fn(member), String(member))
+      }
+   }
+
+   // Booleans
+   await affirmEnchaine([true, false], st.toStringBoolean.bind(st));
+
+   // Bytes
+   await affirmEnchaine([-128, 127], st.toStringI8.bind(rt));
+   await affirmEnchaine([0, 0xFF], st.toStringU8.bind(rt));
+
+   // Shorts
+   await affirmEnchaine([-32768, 32767], st.toStringI16.bind(rt));
+   await affirmEnchaine([0, 0xFFFF], st.toStringU16.bind(rt));
+
+   // Ints
+   await affirmEnchaine([0, 1, -1, -2147483648, 2147483647], st.toStringI32.bind(rt));
+   await affirmEnchaine([0, 0xFFFFFFFF], st.toStringU32.bind(rt));
 
-Assert.ok(true);
+
+   // Longs
+   // NOTE: we cannot represent greater than `Number.MAX_SAFE_INTEGER`
+   await affirmEnchaine([0, 1, -1, Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER], st.toStringI64.bind(rt));
+   await affirmEnchaine([0, Number.MAX_SAFE_INTEGER], st.toStringU64.bind(rt));
+
+   // Floats
+   await affirmEnchaine([0.0, 0.5, 0.25, 1.5], st.toStringFloat.bind(rt));
+
+   // Doubles
+   await affirmEnchaine([0.0, 0.5, 0.25, 1.5], st.toStringDouble.bind(rt));
+
+   // Prove to ourselves that default arguments are being used.
+   // Step 1: call the methods without arguments, and check against the UDL.
+   const op = await Optionneur.init();
+
+   Assert.equal(await op.sinonString(), "default");
+
+   Assert.ok(!await op.sinonBoolean());
+
+   Assert.deepEqual(await op.sinonSequence(), []);
+
+   Assert.equal(await op.sinonNull(), null);
+   Assert.equal(await op.sinonZero(), 0);
+
+
+   // decimal integers
+   Assert.equal(await op.sinonI8Dec(), -42);
+   Assert.equal(await op.sinonU8Dec(), 42);
+   Assert.equal(await op.sinonI16Dec(), 42);
+   Assert.equal(await op.sinonU16Dec(), 42);
+   Assert.equal(await op.sinonI32Dec(), 42);
+   Assert.equal(await op.sinonU32Dec(), 42);
+   Assert.equal(await op.sinonI64Dec(), 42);
+   Assert.equal(await op.sinonU64Dec(), 42);
+
+   // hexadecimal integers
+   Assert.equal(await op.sinonI8Hex(), -0x7f);
+   Assert.equal(await op.sinonU8Hex(), 0xff);
+   Assert.equal(await op.sinonI16Hex(), 0x7f);
+   Assert.equal(await op.sinonU16Hex(), 0xffff);
+   Assert.equal(await op.sinonI32Hex(), 0x7fffffff);
+   Assert.equal(await op.sinonU32Hex(), 0xffffffff);
+   // The following are too big to be represented by js `Number`
+   // Assert.equal(await op.sinonI64Hex(), 0x7fffffffffffffff);
```

Comment:
just want to raise that I came across the problem Ben came across earlier with big numbers